src/backend/joanie/core/api/client.py

```diff
@@ -403,15 +425,14 @@ def abort(self, request, pk=None):  # pylint: disable=no-self-use, invalid-name
                 f'No order found with id "{pk}" owned by {username}.', status=404
             )
 
-        if order.state != enums.ORDER_STATE_PENDING:
-            return Response("Cannot abort a not pending order.", status=403)
+        if order.state == enums.ORDER_STATE_VALIDATED:
+            return Response("Cannot abort a validated order.", status=403)
 
+        order.cancel()
         if payment_id:
             payment_backend = get_payment_backend()
             payment_backend.abort_payment(payment_id)
```

Comment:
I think this code should be in the transition.
Another point is: it's weird that we need a payment id to cancel an order?

Reply:
Payplug need a payment_id to abort a payment.
But the thing is, I don't think we save the payment_id on joanie